accounts-db/src/accounts_db.rs

```diff
@@ -6758,20 +6758,33 @@ impl AccountsDb {
         txn_iter
             .enumerate()
             .map(|(i, txn)| {
-                let account = accounts_and_meta_to_store
-                    .account_default_if_zero_lamport(i)
-                    .map(|account| account.to_account_shared_data())
-                    .unwrap_or_default();
+                let pubkey = accounts_and_meta_to_store.pubkey(i);
+                let account = match accounts_and_meta_to_store.account_default_if_zero_lamport(i) {
+                    Some(account) => {
+                        let account = account.to_account_shared_data();
+                        self.notify_account_at_accounts_update(
+                            slot,
+                            &account,
+                            txn,
+                            pubkey,
+                            &mut write_version_producer,
+                        );
+                        account
+                    }
+                    None => {
+                        let previous_owner = accounts_and_meta_to_store.account(i).owner();
+                        self.notify_account_deleted(
```

Comment:
We can pass the &account same as notify_account_at_accounts_update, so that we do not have to duplicate the special logic of constructing AccountSharedData::default() here and notify_account_deleted implementation.

Reply:
Can be done.